libs/langchain-community/src/vectorstores/opensearch.ts

```diff
@@ -116,21 +121,34 @@ export class OpenSearchVectorStore extends VectorStore {
     );
     const documentIds =
       options?.ids ?? Array.from({ length: vectors.length }, () => uuid.v4());
-    const operations = vectors.flatMap((embedding, idx) => [
-      {
-        index: {
-          _index: this.indexName,
-          _id: documentIds[idx],
+    const operations = vectors.flatMap((embedding, idx) => {
+      const document = [
+        {
+          index: {
+            _index: this.indexName,
+            _id: documentIds[idx],
+          },
         },
-      },
-      {
-        embedding,
-        metadata: documents[idx].metadata,
-        text: documents[idx].pageContent,
-      },
-    ]);
+        {
+          embedding,
+          metadata: documents[idx].metadata,
+          text: documents[idx].pageContent,
+        },
+      ];
+
+      // aoss does not support document id
+      if (this.isAoss) {
+        delete document[0].index?._id;
+      }
```

Comment:
The following errors that occurred when using OpenSearch Serverless have been addressed
```
Document ID is not supported in create/index operation request
```